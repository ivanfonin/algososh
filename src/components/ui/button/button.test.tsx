import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "./button";
import { useState } from "react";

const buttonProps = {
  extraClass: "extra-class",
  text: "Text",
  isLoader: false,
  disabled: false,
};

const ButtonTestComponent = () => {
  const [loader, setLoader] = useState(false);
  const combinedProps = {
    ...buttonProps,
    isLoader: loader,
    onClick: () => setLoader((prevLoader) => !prevLoader),
  };
  return <Button {...combinedProps} />;
};

describe("Проверка снапшота, проверка обработичка событий, состояния загрузки", () => {
  afterEach(cleanup);

  // Сравниваем со снапшотом
  it("Должна матчиться со снапшотом", () => {
    const ButtonComponent = renderer.create(<Button {...buttonProps} />);
    expect(ButtonComponent).toMatchSnapshot();
  });

  // Проверка работоспособности
  it("Должна обрабатывать клик, после клика менять состояние isLoader", async () => {
    render(<ButtonTestComponent />);

    // Находим кнопку
    const buttonElement = screen.getByTestId("button-test");
    // screen.debug(buttonElement);

    // Проверяем текст в кнопке
    expect(buttonElement).toHaveTextContent("Text");

    // Кликаем по кнопке
    fireEvent.click(buttonElement);

    // После клика на кнопке должна появиться иконка загрузки
    expect(screen.getByTestId("svg-loader")).toBeInTheDocument();
  });
});
