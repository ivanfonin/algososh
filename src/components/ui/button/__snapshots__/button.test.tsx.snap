// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент Button Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button-test"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестируем компонент Button Кнопка в состоянии disabled рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button-test"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестируем компонент Button Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button-test"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    data-testid="svg-loader"
    src="loader.svg"
  />
</button>
`;

exports[`Тестируем компонент Button Кнопка с пропсами рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false extra-class"
  data-testid="button-test"
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  >
    Text
  </p>
</button>
`;
